src/main/java/de/tum/in/www1/artemis/domain/notification/SingleUserNotificationFactory.java

```diff
@@ -1,54 +1,64 @@
 package de.tum.in.www1.artemis.domain.notification;
 
 import static de.tum.in.www1.artemis.domain.enumeration.NotificationPriority.*;
+import static de.tum.in.www1.artemis.domain.notification.NotificationConstants.*;
 import static de.tum.in.www1.artemis.domain.notification.NotificationTargetFactory.*;
-import static de.tum.in.www1.artemis.domain.notification.NotificationTitleTypeConstants.*;
 
 import java.util.Set;
 
 import de.tum.in.www1.artemis.domain.Course;
 import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.Lecture;
 import de.tum.in.www1.artemis.domain.User;
 import de.tum.in.www1.artemis.domain.enumeration.NotificationType;
+import de.tum.in.www1.artemis.domain.metis.AnswerPost;
 import de.tum.in.www1.artemis.domain.metis.Post;
 import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismCase;
 import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroup;
 
 public class SingleUserNotificationFactory {
 
-    private static final String POST_NOTIFICATION_TEXT = "Your post got replied.";
-
     /**
      * Creates an instance of SingleUserNotification.
      *
      * @param post             which is answered
+     * @param answerPost       that is replied with
      * @param notificationType type of the notification that should be created
      * @param course           that the post belongs to
      * @return an instance of SingleUserNotification
      */
-    public static SingleUserNotification createNotification(Post post, NotificationType notificationType, Course course) {
+    public static SingleUserNotification createNotification(Post post, AnswerPost answerPost, NotificationType notificationType, Course course) {
```

Comment:
This entire code is duplicated in GroupNotificationFactory. Please refactor it otherwise we will always have to apply changes in two places.

Reply:
I agree, this could be part of some NotificationBuilder class

Reply:
You are right. There is some similarity in the structure of the code, but even though they get triggered by the same event they are different notifications. One is for the author of the post that gets replied to and one is for the whole course. They have different texts and we should keep the possibility to provide different placeholder values as well. I refactored the placeholder generation in a separate method to reduce the code duplication but I don't think we should mix the generation of the SingleUser and GroupNotification.

Reply:
@TimOrtel I generally agree, but I think that's a problem with the NotificationFactorys in general and not related to this changes. I think this should be refactored in a seperate iteration